### PR TITLE
Feature/upgrade entities

### PR DESCRIPTION
**Pull request by zeroZshadow**
#### Description
Upgrade the FPS to use the new Entities API's.

#### Tests
Ran locally in every scene.
Cloud deployment is currently launching

#### Documentation
Changelog updated.

### CODE REVIEW

File: workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs [hunk 1]
```diff
@@ -45,7 +46,10 @@ protected override void OnUpdate()
             }
             else
             {
-                SessionStatus = sessionGroup.GetComponentDataArray<Session.Component>()[0].Status;
+                Entities.With(sessionGroup).ForEach((ref Session.Component session) =>
```
Comment (paulbalaji) [inline workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs]: is there any chance that we have more than 1 entity matching this query?

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 2]
```diff
@@ -131,75 +126,63 @@ private void ApplyHealthRegen()
...
-            for (var i = 0; i < spatialIdData.Length; i++)
-            {
-                var healthComponent = healthComponentData[i];
-                var regenComponent = healthRegenComponentData[i];
-                var regenData = healthRegenData[i];
-
-                // Don't regen if dead.
-                if (healthComponent.Health == 0)
+            Entities.With(regenGroup).ForEach(
+                (ref SpatialEntityId spatialEntityId,
```
Comment (paulbalaji) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: nit: indenting of first one is different

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs [hunk 3]
```diff
@@ -131,75 +126,63 @@ private void ApplyHealthRegen()
...
+                    // If not damaged recently, and not already fully healed, regen.
+                    if (healthComponent.Health < healthComponent.MaxHealth)
+                    {
+                        regenData.NextRegenTimer -= Time.deltaTime;
+                        if (regenData.NextRegenTimer <= 0)
+                        {
+                            regenData.NextRegenTimer += regenComponent.RegenInterval;
+
+                            // Send command to regen entity.
+                            var modifyHealthRequest = new HealthComponent.ModifyHealth.Request(
```
Comment (paulbalaji) [inline workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: should we be doing
```csharp
.Request(
	///
);
```

or

```csharp
.Request
(
	///
);
```

File: workers/unity/ProjectSettings/QualitySettings.asset [hunk 4]
```diff
@@ -123,7 +123,7 @@ QualitySettings:
     shadowCascade2Split: 0.33333334
     shadowCascade4Split: {x: 0.06666667, y: 0.2, z: 0.46666667}
     shadowmaskMode: 0
-    blendWeights: 1
+    skinWeights: 1
```
Comment (paulbalaji) [inline workers/unity/ProjectSettings/QualitySettings.asset]: why did the name change?

File: workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs (hunk 1 above)
Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs]: if you misconfigure it >< 
but you should always have at most one session entity in your deployment.

File: workers/unity/Assets/Fps/Scripts/SetupLogic/OneTimeInitialisation.cs [hunk 5]
```diff
@@ -18,8 +16,6 @@ private static void Init()
             }
 
             initialized = true;
-            WorldsInitializationHelper.SetupInjectionHooks();
-            PlayerLoopManager.RegisterDomainUnload(WorldsInitializationHelper.DomainUnloadShutdown, 1000);
```
Comment (jessicafalk) [inline workers/unity/Assets/Fps/Scripts/SetupLogic/OneTimeInitialisation.cs]: that not needed anymore? (the domainunloadshutdown bit)

File: workers/unity/ProjectSettings/QualitySettings.asset (hunk 4 above)
Comment (zeroZshadow) [reply workers/unity/ProjectSettings/QualitySettings.asset]: Probably re-exported when I build out a client. Upgrade to 2019.1 still had some name changes left.

File: workers/unity/Assets/Fps/Scripts/SetupLogic/OneTimeInitialisation.cs (hunk 5 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Fps/Scripts/SetupLogic/OneTimeInitialisation.cs]: You're right, deleted a line too many. Putting this back.

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs (hunk 2 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: This was done by our formatting rules

File: workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs (hunk 3 above)
Comment (zeroZshadow) [reply workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs]: Its formatted by the formatter this way.

File: workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs (hunk 1 above)
Comment (zeroZshadow) [reply workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs]: @jessicafalk I think no?

Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs]: as mentioned: our snapshot always only contains at most one entity. A user can of course hack around with it and change it, but we assume you only have it at most once

Comment (zeroZshadow) [reply workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs]: Is it oke by you to leave it like this?

Comment (jessicafalk) [reply workers/unity/Assets/Fps/Scripts/GameLogic/Session/TrackPlayerSystem.cs]: it's not nice, but ok to leave it like this for now. It's not a big issue

**Pull request merged**